shm/map.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include <vector>
+#include <memory>
+#include <functional>
+#include <algorithm>
+
+class Map {
+public:
+    Map();
+    Island* const getPosition();
+    Island* getIsland(const Island::Coordinates& coordinate);
+private:
+    const int maxWidth_ = 99;
+    const int maxHeight_ = 99;
+    const int islandsQuantity_ = 10;
```

Comment:
Better mark it as static constepxr or if you need it only in .cpp file move it there to anonymous namespace